tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Http/WebRequest/HttpWebRequest_EndGetResponse_Integration.cs

```diff
@@ -94,9 +94,22 @@ public class HttpWebRequest_EndGetResponse_Integration
                         if (returnValue is HttpWebResponse response)
                         {
                             scope.Span.SetHttpStatusCode((int)response.StatusCode, isServer: false, Tracer.Instance.Settings);
+                            scope.Dispose();
                         }
+                        else if (exception is WebException { Status: WebExceptionStatus.ProtocolError, Response: HttpWebResponse exceptionResponse })
+                        {
+                            scope.Span.SetHttpStatusCode((int)exceptionResponse.StatusCode, isServer: false, Tracer.Instance.Settings);
 
-                        scope.DisposeWithException(exception);
+                            // SetHttpStatusCode will mark the span with an error if the StatusCode is within the configured range
+                            // Add the remaining error attributes without setting the Error property
+                            scope.Span.SetExceptionTags(exception);
```

Comment:
nit: I think we should reverse the order of these? `SetHttpStatusCode()` checks to see if there's already an error code set, and if not writes a placeholder one:

```csharp
if (string.IsNullOrEmpty(span.GetTag(Tags.ErrorMsg)))
{
    span.SetTag(Tags.ErrorMsg, $"The HTTP response has status code {statusCodeString}.");
}
```

By reversing the order we avoid the unnecessary `SetTag()` (as we fill it with the exception tag). Same overall outcome, just one less "SetTag". 

Alternatively, we _could_ add an overload to `SetHttpStatusCode` that also takes an exception and avoids the `GetTag` call too, but probably unnecessary if this behaviour only applies to `WebRequest`

Reply:
Good idea with reversing the order, done in c909a6a0939f533badc2944484250e63b64b9311